code/physics/physics.cpp

```diff
@@ -1171,6 +1187,44 @@ void physics_add_point_mass_moi(matrix *moi, float mass, vec3d *pos)
 	moi->a2d[2][2] += mass * (pos->xyz.x * pos->xyz.x + pos->xyz.y * pos->xyz.y);
 }
 
+// equation from https://en.wikipedia.org/wiki/Projectile_motion#Angle_%CE%B8_required_to_hit_coordinate_(x,_y)
+bool physics_lead_ballistic_trajectory(const vec3d* start, const vec3d* end_pos, const vec3d* target_vel, float weapon_speed, const vec3d* gravity, vec3d* out_direction) {
+	float best_guess_time = 0.0f;
+	*out_direction = vmd_zero_vector;
+	float time = 0.0f;
+	for (int i = 0; i < 6; i++) {
```

Comment:
Why is this iterating 6 times?

Reply:
It is an iterative solution, and while I don't think it's all that expensive, it's good to have some kind of upper limit.